cmake/OpenSSL.cmake

```diff
@@ -1,15 +1,34 @@
+# Check if a file is a symbolic link (macOS)
+function(is_symlink filepath result)
+    execute_process(
+        COMMAND "test" "-h" "${filepath}"
+        RESULT_VARIABLE return_code
+    )
+    if(return_code EQUAL 0)
+        set(${result} TRUE PARENT_SCOPE)
+    else()
+        set(${result} FALSE PARENT_SCOPE)
+    endif()
+endfunction()
+
 if(NOT DEFINED OPENSSL_CONFIGURED)
     if(APPLE)
-        execute_process(COMMAND brew --prefix openssl@3.1
-                OUTPUT_VARIABLE OPENSSL_PREIX RESULT_VARIABLE OPENSSL_RC
+        execute_process(COMMAND brew --prefix openssl@3
+                OUTPUT_VARIABLE OPENSSL_PREFIX RESULT_VARIABLE OPENSSL_RC
                 OUTPUT_STRIP_TRAILING_WHITESPACE)
 
         if(NOT OPENSSL_RC EQUAL 0)
             message(FATAL_ERROR "No OpenSSL found, use homebrew to install one")
         endif()
-
-        message(STATUS "Found OpenSSL at ${OPENSSL_PREIX}")
-        set(OPENSSL_ROOT_DIR ${OPENSSL_PREIX} CACHE INTERNAL "OpenSSL")
+        is_symlink(${OPENSSL_PREFIX} is_symlink_result)
+        message(STATUS "Found OpenSSL at ${OPENSSL_PREFIX}")
+        if(is_symlink_result)
+            file(REAL_PATH ${OPENSSL_PREFIX} absolute_path)
+            set(OPENSSL_ROOT_DIR ${absolute_path} CACHE INTERNAL "OpenSSL")
+        else()
+            set(OPENSSL_ROOT_DIR ${OPENSSL_PREFIX} CACHE INTERNAL "OpenSSL")
+        endif()
+        message(STATUS " OpenSSL root dir at ${OPENSSL_ROOT_DIR}")
```

Comment:
Any particular reason to save the absolute path? Is it so that if it changes over time (homebrew updates openssl@3 to point to 3.3 or whatever), we still use the version we picked previously?

Would be prudent to document the reasoning here.

Reply:
I overlooked it. No need to cache that value. Removed in the latest commit.